services/workflows-service/src/statistics/statistics.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '@/prisma/prisma.service';
+import { StatisticsOutputDto } from '@/statistics/statistics-output.dto';
+import { StatisticsInputDto } from '@/statistics/statistics-input.dto';
+import type { TProjectId, TProjectIds } from '@/types';
+import { ApprovalState, BusinessReportStatus } from '@prisma/client';
+
+const LOW_LTE_RISK_SCORE = 39;
+const MEDIUM_LTE_RISK_SCORE = 69;
+const HIGH_LTE_RISK_SCORE = 84;
+const CRITICAL_GTE_RISK_SCORE = 85;
+
+@Injectable()
+export class StatisticsService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async getStatistics(
+    input: StatisticsInputDto,
+    projectIds: TProjectIds,
+  ): Promise<StatisticsOutputDto> {
+    return {
+      riskIndicators: await this.getRiskIndicators(projectIds![0]!),
+      reports: {
+        all: await this.getReportsByRiskLevel(projectIds![0]!),
+        inProgress: await this.getInProgressReportsByRiskLevel(projectIds![0]!),
+        approved: await this.getApprovedBusinessesReportsByRiskLevel(projectIds![0]!),
+      },
+    };
+  }
+
+  async getRiskIndicators(projectId: TProjectId) {
+    return (
+      await this.prismaService.$queryRaw<Array<{ name: string; count: string }>>`
+        WITH "flattenedRiskIndicators" AS (SELECT jsonb_array_elements("report" -> 'data' -> 'summary' ->
+          'riskIndicatorsByDomain' ->
+          (jsonb_object_keys("report" -> 'data' -> 'summary' -> 'riskIndicatorsByDomain'))) AS "riskIndicator",
+          "projectId"
+        FROM
+          "BusinessReport"
+          )
+        SELECT
+          "riskIndicator" ->> 'name' AS name, COUNT(*) AS count
+        FROM
+          "flattenedRiskIndicators"
+        WHERE
+          "riskIndicator" ->> 'name' IS NOT NULL
+          AND "projectId" = ${projectId}
+        GROUP BY
+          "riskIndicator" ->> 'name'
+        ORDER BY
+          "count" DESC,
+          "riskIndicator" ->> 'name' ASC;`
+    ).map(({ name, count }) => ({
+      name,
+      count: Number(count),
+    }));
+  }
+
+  async getReportsByRiskLevel(projectId: TProjectId) {
+    const results = await this.prismaService.$queryRaw<
+      Array<{ riskLevel: 'low' | 'medium' | 'high' | 'critical'; count: number }>
```

Comment:
We have a type in `@ballerine/common` called `TSeverity` and a `Severity` enum.

Reply:
It's in the backoffice, not the `@ballerine/common`.